arch/arm/core/exc_exit.S

```diff
@@ -69,13 +70,16 @@ SECTION_SUBSEC_FUNC(TEXT, _HandlerModeExit, _IntExit)
  */
 
 SECTION_SUBSEC_FUNC(TEXT, _HandlerModeExit, _ExcExit)
+#ifdef CONFIG_CPU_CORTEX_R
+    push {r0}
+#endif
 
 #ifdef CONFIG_PREEMPT_ENABLED
     ldr r0, =_kernel
 
     ldr r1, [r0, #_kernel_offset_to_current]
 
-    ldr r0, [r0, _kernel_offset_to_ready_q_cache]
+    ldr r0, [r0, #_kernel_offset_to_ready_q_cache]
```

Comment:
What was the reason for this?

Reply:
Using the zephyr-sdk, I get a compile error
src/zephyr/arch/arm/core/exc_exit.S: Assembler messages:
src/zephyr/arch/arm/core/exc_exit.S:82: Error: immediate expression requires a # prefix -- `ldr r0,[r0,(0x1c+0x0)]' since _kernel_offset_to_ready_q_cache is an immediate value.  I can't really say why the M boards do not see this.